src/main/java/com/academy/ui/pages/greenCity/BasePageGreenCity.java

```diff
@@ -0,0 +1,20 @@
+package com.academy.ui.pages.greenCity;
+
+import com.academy.ui.components.HeaderComponent;
+import com.academy.ui.components.RegistrationModalComponent;
+import com.academy.ui.pages.BasePage;
+import org.openqa.selenium.WebDriver;
+
+public class BasePageGreenCity extends BasePage {
+    public BasePageGreenCity(WebDriver driver) {
+        super(driver);
+    }
+
+    public HeaderComponent getHeaderComponent() {
+        return new HeaderComponent(driver, headerRootElement);
```

Comment:
Wouldn't it be better to add the HeaderComponent as a class field and define it within constructor. This way, the HeaderComponent will  be created not only when the getHeaderComponent method is invoked. The same for Ubs Base Page